src/game-logic/reducers/addItemToInventory.js

```diff
@@ -0,0 +1,43 @@
+import { inventorySpaceRemaining } from '../../utils'
+
+/**
+ * Only adds as many items as there is room in the inventory for unless
+ * allowInventoryOverage is true.
+ * @param {farmhand.state} state
+ * @param {farmhand.item} item
+ * @param {number} [howMany=1]
+ * @param {boolean} [allowInventoryOverage=false]
+ * @returns {farmhand.state}
+ */
+export const addItemToInventory = (
```

Comment:
what do you think about making these default exports? I still tend to use the `function` keyword though I have no strong opinions one way or another on that specifically (but I will use it in my example hah). i think making these the default export makes sense though since they are the only export, i.e. `export default function addItemToInventory(...`

the `index.js` can also take advantage of this with a one line import/export like `export { default as addItemToInventory } from "./addItemToInventory"`.

Reply:
It was a conscious choice to avoid default exports here, as I'm seeing them more and more as an antipattern. [This article](https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad) makes a case against them. The point I find compelling in that article is the "Refactoring" section:

> Default exports make large-scale refactoring impossible since each importing site can name default import differently (including typos).

```js
// in file exports.js
export default function () {...}// in file import1.js

import doSomething from "./exports.js"// in file import2.js

import doSmth from "./exports.js"
```

So, relying solely on named exports seems like a good syntactical guard against human error and the maintenance challenges that can result.

I realize that default exports are currently used throughout the codebase, and updating them all would be at least [a modest effort](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/jeremyckahn/farmhand%24+export+default&patternType=literal). Rather than going back to fix them all, I could draft an ADR to define the project standard for new code to adhere to. We could even use [this ESLint rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-default-export.md) to enforce style once we have removed them all.

How would you feel about setting this code standard for the project?

Reply:
thanks for linking the article! the tree-shaking seems like a really great reason too. i've never read this, but i'm definitely ok with setting this as a standard for this project. i'll have to ponder a bit more before i go to work and change our stance, but you might have me sold 😄

Reply:
I've added an ADR for this decision in https://github.com/jeremyckahn/farmhand/commit/6bdc2cd3b2c11522e0f95f3e1fa6f9f0c4df0b53!